### ISSUE TITLE
404 Page Causes Infinite Reload After Using `npx create-remix@latest`

### ISSUE THREAD

**Issue by y0tik** [remix-run/remix#9821]
### Reproduction

1. Install the latest version using one of the following methods:
   - With `npx create-remix@latest`
   - Or goto https://stackblitz.com/edit/remix-run-remix-rwkrfb

2. Start the development server

3. Navigate to [http://localhost:5173/404](http://localhost:5173/404) or any other non-existent page

### System Info

```shell
System:
    OS: Linux 5.15 Arch Linux
    CPU: (16) x64 12th Gen Intel(R) Core(TM) i5-12500H
    Memory: 6.89 GB / 7.61 GB
    Container: Yes
    Shell: 3.7.1 - /bin/fish
  Binaries:
    Node: 20.14.0 - ~/.local/share/nvm/v20.14.0/bin/node
    npm: 10.7.0 - ~/.local/share/nvm/v20.14.0/bin/npm
    pnpm: 9.3.0 - ~/.local/share/pnpm/pnpm
```

### Used Package Manager

pnpm

### Expected Behavior

When visiting a 404 page, display the nearest or the root ErrorBoundary without causing an infinite reload

### Actual Behavior

Infinity page reload when visiting a 404 page, both in development and in production

**Comment by rossipedia** [comment]
FWIW I cannot reproduce this using v2.10.3, but I can with v2.11.0

**Comment by bravo-kernel** [comment]
My codebase runs 2.10.3 and has the same issue: 

- Infinite 404 loop)
- Stops when I disable the `<Scripts />` component in my layout

**Comment by Arichy** [comment]
Same here.
<img width="525" alt="image" src="https://github.com/user-attachments/assets/2510348d-6ab2-4ae9-8b38-ccbcbc6f6ad1">

After downgrading `@remix-run/serve` from `2.11.0` to `2.10.3`, it's fixed. Look like something is wrong in latest version.
<img width="382" alt="image" src="https://github.com/user-attachments/assets/be464069-8754-4aa6-952b-567eb1a4f91c">

**Comment by alcpereira** [comment]
To help narrow down the issue:
- Only happens in `dev` server
- Only happens in `2.11.0`
- I tried reverting commit [`ef612a729bf6acd6b9a11ba6c3971beb1f1760f1`](https://github.com/remix-run/remix/commit/ef612a729bf6acd6b9a11ba6c3971beb1f1760f1) as a first guess, without success

**Comment by ironbyte** [comment]
Yep, can confirm I ran into this very issue today. I had to downgrade to 2.10.x which doesn't have this bug.

**Comment by ngbrown** [comment]
It also happens on production servers built and deployed to Cloudflare.

**Comment by petrpacas** [comment]
Happens for me in both dev and prod with 2.11.0, doesn't happen with 2.10.3

Note to self: Remember you bumped packages before you encountered weird error and try to debug it

**Comment by machour** [comment]
This should be fixed by [v2.11.1](https://github.com/remix-run/remix/blob/main/CHANGELOG.md#v2111).
Could you all confirm? 🙏🏼

**Comment by petrpacas** [comment]
Fixed for me
